stl/inc/xutility

```diff
@@ -3148,6 +3174,21 @@ namespace ranges {
     // VARIABLE ranges::prev
     inline constexpr _Prev_fn prev{_Not_quite_object::_Construct_tag{}};
 
+    // FUNCTION TEMPLATE _Find_last_iterator
+    template <forward_iterator _It, sentinel_for<_It> _Se>
+    _NODISCARD constexpr _It _Find_last_iterator(
+        const _It& _First, const _Se& _Last, const iter_difference_t<_It> _Count) {
+        // Find the iterator in [_First, _Last) (of length _Count) which equals _Last
+        _STL_INTERNAL_CHECK(_RANGES distance(_First, _Last) == _Count);
+        auto _Result = _First;
+        if constexpr (assignable_from<_It&, _Se>) {

```

Comment:
This is exactly the requirement from [[range.iter.op.advance]/4.1](http://eel.is/c++draft/range.iter.op.advance#4.1). The code is asking if the advance call will use the bogus<sup>1</sup> assignment optimization path. I should walk this back to `same_as<_It, Se>` and perform the assignment directly here.

<sup>1</sup> [range.iter.op.advance]/4.1 is fine for callers of `advance` with concrete types who see the effects and choose to make the call or not, but since it performs an operation not covered by the `input_or_output_iterator` or `sentinel_for` requirements generic code effectively can't call that `advance` overload if it might perform that operation since the semantics are completely unknown. This paragraph has bothered me for literally years and I've only just now _finally_ figured out why. This needs deserves an STL issue, and an LWG issue, and a fix.